src/ZuzaluOracle.sol

```diff
@@ -1,65 +1,161 @@
 pragma solidity 0.8.4;
 
-import { SemaphoreVerifier } from "@semaphore-protocol/contracts/base/SemaphoreVerifier.sol";
-import { Owned } from "solmate/auth/Owned.sol";
+import {ISemaphoreVerifier} from "@semaphore-protocol/contracts/interfaces/ISemaphoreVerifier.sol";
+import {Owned} from "solmate/auth/Owned.sol";
 
-/// @title
+/// @title A wrapper contract around the Semaphore Verifier that is used by Zuzalu to verify it's members
+/// @author Mark Tyneway <mark.tyneway@gmail.com>
+/// @author Odysseas.eth <odyslam@gmail.com>
 contract ZuzaluOracle is Owned {
-    /// @notice
-    struct Commitment {
-        uint256 root;
-        uint256 depth;
+    /// The official groups by Zuzalu as defined and used in the backend
+    enum Groups
+    // Dummy value so that groups have the official numbering (1-4)
+    {
+        None,
+        Participants,
+        Residents,
+        Visitors,
+        Organizers
     }
 
-    /// @notice
-    event Update(uint256 root, uint256 depth);
+    /// @notice The groups have been updated with new latest roots and depths
+    event UpdateGroups(uint256[4] roots, uint256[4] depths);
+    /// @notice A new succesful verification has been made for a particular Zuzalu group and signal
+    event Verify(uint256 indexed signal, Groups indexed _group);
 
-    /// @notice
-    error NotCanonical(uint256);
+    /// @notice The group is not one of the groups in the Groups enum
+    error InvalidGroup();
 
-    /// @notice
-    Commitment private _latestCommitment;
+    /// @notice An array of roots for the "visitors" group
+    uint256[] $visitorRoots;
+    /// @notice An arry of roots for the "residents" group
+    uint256[] $residentRoots;
+    /// @notice An array of roots for the "organizers" group
+    uint256[] $organizerRoots;
+    /// @notice An array of roots for the "participants" group
+    uint256[] $participantRoots;
 
-    /// @notice root -> depth
-    mapping(uint256 => uint256) public commitments;
+    /// @notice A mapping of roots to their depth for the "visitors" groups
+    mapping(uint256 => uint256) public $visitorsToDepth;
+    /// @notice A mapping of roots to their depth for the "residents" groups
+    mapping(uint256 => uint256) public $residentsToDepth;
+    /// @notice A mapping of roots to their depth for the "organizers" groups
+    mapping(uint256 => uint256) public $organizersToDepth;
+    /// @notice A mapping of roots to their depth for the "participants" groups
+    mapping(uint256 => uint256) public $participantsToDepth;
 
-    /// @notice
-    address immutable public VERIFIER;
+    /// @notice The address of the Semaphore verifier contract
+    address public immutable VERIFIER;
 
-    /// @notice
-    constructor(Commitment memory _commitment, address _owner, address _verifier) Owned(_owner) {
+    constructor(address _owner, address _verifier) Owned(_owner) {
         VERIFIER = _verifier;
-        _update(_commitment);
+        _initArrays();
     }
 
```

Comment:
Do you init the arrays so that the getter doesn't revert?

Reply:
Yes, it seemed cleaner than adding special code

Reply:
Perhaps comment reasoning